examples/gno.land/p/demo/grc/grc721/grc721_royalty.gno

```diff
@@ -0,0 +1,78 @@
+package grc721
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+)
+
+// royaltyNFT represents a non-fungible token (NFT) with royalty functionality.
+type royaltyNFT struct {
+	*metadataNFT          // Embedding metadataNFT for NFT functionality
+	tokenRoyaltyInfo avl.Tree  // AVL tree to store royalty information for each token
+}
+
+// Ensure that royaltyNFT implements the IGRC2981 interface.
+var _ IGRC2981 = (*royaltyNFT)(nil)
+
+// NewNFTWithRoyalty creates a new royalty NFT with the specified name, symbol, and royalty calculator.
+func NewNFTWithRoyalty(name string, symbol string) *royaltyNFT {
+	// Create a new NFT with metadata
+	nft := NewNFTWithMetadata(name, symbol)
+
+	return &royaltyNFT{
+		metadataNFT:        nft,
+		tokenRoyaltyInfo:   avl.Tree{},
```

Comment:
Nit: use the `avl.NewTree()` function

Reply:
https://github.com/linhpn99/gno/commit/c035de8d73314aad8969b16107d49db0a7fdb882